File: springapp/CRMS_Frontend/src/services/DepartmentService.js
import api from '../utils/api';

const DepartmentService = {
    getAll: async () => {
        const response = await api.get('/departments');
        return response.data;
    },
    getById: async (id) => {
        const response = await api.get(`/departments/${id}`);
        return response.data;
    },
    create: async (dept) => {
        const response = await api.post('/departments', dept);
        return response.data;
    },
    update: async (id, dept) => {
        const response = await api.put(`/departments/${id}`, dept);
        return response.data;
    },
    delete: async (id) => {
        await api.delete(`/departments/${id}`);
    }
};

export default DepartmentService;

File: springapp/CRMS_Backend/src/main/java/com/examly/springapp/service/DepartmentService.java
package com.examly.springapp.service;

import java.util.List;
import java.util.Optional;

import org.springframework.data.domain.Page;

import com.examly.springapp.model.Department;

public interface DepartmentService {

    Department create(Department dept);

    List<Department> read();

    Optional<Department> getById(Long id);

    Department update(Department dept);

    void delete(Long id);

    Page<Department> getDepartments(int page, int size);
}